rialto_airflow/publish/publication.py

```diff
@@ -0,0 +1,313 @@
+                    func.jsonb_agg_strict(Author.primary_role).label("primary_role"),
+                    func.jsonb_agg_strict(Funder.name).label("funders"),
+                    func.jsonb_agg_strict(Funder.federal).label("federal"),
+                )
+                .join(Author, Publication.authors)  # type: ignore
+                .group_by(Publication.id)
+                .execution_options(yield_per=100)
+            )
+
+            for row in session.execute(stmt):
+                csv_output.writerow(
+                    {
+                        "doi": row.doi,
+                        "pub_year": row.pub_year,
+                        "apc": row.apc,
+                        "open_access": row.open_access,
+                        "types": "|".join(get_types(row)) or None,
+                        "federally_funded": any(row.federal),
+                        "academic_council_authored": any(row.academic_council),
+                        "faculty_authored": "faculty" in row.primary_role,
+                    }
+                )
+
+        logging.info(f"finished writing publications {csv_path}")
+
+    return csv_path
+
+
+def write_contributions_by_school(snapshot) -> Path:
+    """
+    Write a CSV of contributions where each row represents a unique publication per school.
+    """
+
+    col_names = [
+        "academic_council_authored",
+        "academic_council",
+        "journal",
+        "issue",
+        "mesh",
+        "pages",
+        "volume",
+        "apc",
+        "doi",
+        "pmid",
+        "title",
+        "url",
+        "faculty_authored",
+        "federally_funded",
+        "open_access",
+        "primary_school",
+        "primary_department",
+        "role",
+        "sunet",
+        "pub_year",
+        "types",
+    ]
+
+    csv_path = get_csv_path(
+        snapshot, google_drive_folder(), "contributions-by-school.csv"
+    )
+
+    logging.info(f"starting to write contributions by school {csv_path}")
+
+    with csv_path.open("w") as output:
+        csv_output = DictWriter(output, fieldnames=col_names)
+        csv_output.writeheader()
+
+        with get_session(snapshot.database_name).begin() as session:
+            stmt = (
+                select(  # type: ignore
+                    Publication.apc,  # type: ignore
+                    Publication.doi,  # type: ignore
+                    Publication.title,  # type: ignore
+                    Publication.open_access,  # type: ignore
+                    Publication.dim_json,
+                    Publication.openalex_json,
+                    Publication.sulpub_json,  # type: ignore
+                    Publication.wos_json,  # type: ignore
+                    Publication.pubmed_json,  # type: ignore
+                    Author.primary_school,  # type: ignore
+                    Author.primary_dept,  # type: ignore
+                    Author.primary_role,  # type: ignore
+                    Author.sunet,  # type: ignore
```

Comment:
If the requirement is to summarize publications by school I think we will want to remove Author.sunet, Author.primary_role, and Author.primary_dept from the columns being returned?